database/rrddim.c

```diff
@@ -307,11 +309,20 @@ RRDDIM *rrddim_add_custom(RRDSET *st, const char *id, const char *name, collecte
                     memset(rd, 0, size);
                     reset = 1;
                 }
-                else if(dt_usec(&now, &rd->last_collected_time) > (rd->entries * rd->update_every * USEC_PER_SEC)) {
+                // Collected times are valid on the collecting node
+                else if(host == localhost &&
+                    dt_usec(&now, &rd->last_collected_time) > (rd->entries * rd->update_every * USEC_PER_SEC)) {
```

Comment:
You should probably deduplicate `dt_usec(&now, &rd->last_collected_time) > (rd->entries * rd->update_every * USEC_PER_SEC)` by putting its value in a variable.

Reply:
ah, scratch that, I see that it's different after all.